### PR TITLE
Adding CSS classes to pagebreak.php and other improvements

### PR DESCRIPTION
**Pull request by AndySDH**
Updated Pull Request for Issue #14670, now that I got more knowledge on how a PR is made correctly.

Since the navigation for multi-page articles (plugins/content/pagebreak.pagebreak.php) can't be template-overridden (unlike the article navigation in pagenavigation.php, which can), we should make some basic improvements to it in the core. This is what I did for my site and would be beneficial in general:

### Summary of Changes

This PR:

- Adds CSS classes to the prev/next links, so they can be stylized via CSS (currently they have no class)

- Adds a pagenavcounter also at the bottom of the page.

- Changes behavior so that the "Previous" Link is not shown when we are on the first page, and the "Next" Link is not shown when we are on the last page. When we are on the first page, there is little point in showing a blank unclickable "prev" link. Sites don't show that. Same thing for when we are on the last page, the "next link" should not be shown. 

- Adds a different language string for these next page/prev page links, as they are not the generic next/prev article links, but they represent next page/previous page. By having a different language string we can individually value these as "PREV PAGE" and "NEXT PAGE".

### Documentation Changes Required

None.

### CODE REVIEW

File: plugins/content/pagebreak/pagebreak.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ public function onContentPrepare($context, &$row, &$params, $page = 0)
 			{
 				throw new Exception(JText::_('JERROR_PAGE_NOT_FOUND'), 404);
 			}
-
+
```
Comment (C-Lodder) [inline plugins/content/pagebreak/pagebreak.php]: remove tabs

File: plugins/content/pagebreak/pagebreak.php [hunk 2]
```diff
@@ -355,27 +358,19 @@ protected function _createNavigation(&$row, $page, $n)
...
+			$next = '<a class="next_page" href="' . $link_next . '">' . JText::_( 'Next Page' ) . $pnSpace . JText::_('JGLOBAL_GT') . '</a>';
 		}
 
 		if ($page > 0)
 		{
 			$page_prev = $page - 1 === 0 ? '' : $page - 1;
 
 			$link_prev = JRoute::_(ContentHelperRoute::getArticleRoute($row->slug, $row->catid, $row->language) . '&showall=&limitstart=' . $page_prev);
-
+
```
Comment (C-Lodder) [inline plugins/content/pagebreak/pagebreak.php]: remove tab

File: plugins/content/pagebreak/pagebreak.php [hunk 3]
```diff
@@ -355,27 +358,19 @@ protected function _createNavigation(&$row, $page, $n)
...
-			$prev = '<a href="' . $link_prev . '">' . JText::_('JGLOBAL_LT') . JText::_('JGLOBAL_LT') . $pnSpace . JText::_('JPREV') . '</a>';
-		}
-		else
-		{
-			$prev = JText::_('JPREV');
+			$prev = '<a class="prev_page" href="' . $link_prev . '">' . JText::_('JGLOBAL_LT') . $pnSpace . JText::_( 'Prev Page' ) . '</a>';
 		}
 
-		$row->text .= '<ul><li>' . $prev . ' </li><li>' . $next . '</li></ul>';
+	  $row->text .= $prev . $next;
```
Comment (C-Lodder) [inline plugins/content/pagebreak/pagebreak.php]: remove space before `$row`

**Comment by C-Lodder** [comment]
Not sure about others but I dont agree with this. I don't see the point in adding classes if they're not actually going to be used in core.

Instead I think the HTML markup should be moved to a layout so it can be overridden and changed by a user, **if** need be.

**Comment by mbabker** [comment]
> Instead I think the HTML markup should be moved to a layout so it can be overridden and changed by a users if need be.

^^ That

**Comment by AndySDH** [comment]
> Not sure about others but I dont agree with this. I don't see the point in adding classes if they're not actually going to be used in core.

Note that this doesn't just add classes but also makes a lot of other improvements, including change to if conditionals when on first/last page, and new language strings specific for the occurrence.

But feel free to turn it into a layout or to do something different if you wish!

**Comment by mbabker** [comment]
It's not necessarily that the changes here are bad, but if you've got a situation where you need to adjust the markup the only option is to duplicate the plugin and maintain it all because the markup is hardcoded into it.

Although to be honest this is one of the plugins I just want to get rid of in 4.0 and just merge all the logic into com_content.  Because this plugin literally only supports the article view of that component.

File: plugins/content/pagebreak/pagebreak.php (hunk 1 above)
Comment (Quy) [inline plugins/content/pagebreak/pagebreak.php:83]: Remove tabs

File: plugins/content/pagebreak/pagebreak.php [hunk 4]
```diff
@@ -355,27 +358,19 @@ protected function _createNavigation(&$row, $page, $n)
...
+			$next = '<a class="next_page" href="' . $link_next . '">' . JText::_( 'JNEXT_PAGE' ) . $pnSpace . JText::_('JGLOBAL_GT') . '</a>';
 		}
 
 		if ($page > 0)
 		{
 			$page_prev = $page - 1 === 0 ? '' : $page - 1;
 
 			$link_prev = JRoute::_(ContentHelperRoute::getArticleRoute($row->slug, $row->catid, $row->language) . '&showall=&limitstart=' . $page_prev);
-
+
```
Comment (Quy) [inline plugins/content/pagebreak/pagebreak.php:369]: Remove tab

File: plugins/content/pagebreak/pagebreak.php [hunk 5]
```diff
@@ -355,27 +358,19 @@ protected function _createNavigation(&$row, $page, $n)
...
-			$prev = '<a href="' . $link_prev . '">' . JText::_('JGLOBAL_LT') . JText::_('JGLOBAL_LT') . $pnSpace . JText::_('JPREV') . '</a>';
-		}
-		else
-		{
-			$prev = JText::_('JPREV');
+			$prev = '<a class="prev_page" href="' . $link_prev . '">' . JText::_('JGLOBAL_LT') . $pnSpace . JText::_( 'JPREV_PAGE' ) . '</a>';
 		}
 
-		$row->text .= '<ul><li>' . $prev . ' </li><li>' . $next . '</li></ul>';
+	  $row->text .= $prev . $next;
```
Comment (Quy) [inline plugins/content/pagebreak/pagebreak.php:374]: Use tab

File: language/en-GB/en-GB.ini [hunk 6]
```diff
@@ -87,6 +88,7 @@ JPAGETITLE="%1$s - %2$s"
 JPREV="Prev"
 JPREVIOUS="Previous"
 JPREVIOUS_TITLE="Previous article: %s"
+JPREV_PAGE="Prev Page"
```
Comment (Quy) [inline language/en-GB/en-GB.ini:91]: Sort key in alpha order

File: language/en-GB/en-GB.ini [hunk 7]
```diff
@@ -75,6 +75,7 @@ JMONTH="Month"
 JNEW="New"
 JNEXT="Next"
 JNEXT_TITLE="Next article: %s"
+JNEXT_PAGE="Next Page"
```
Comment (Quy) [inline language/en-GB/en-GB.ini:78]: Sort key in alpha order

File: administrator/language/en-GB/en-GB.ini [hunk 8]
```diff
@@ -102,6 +103,7 @@ JONLY="Only"
 JOPTIONS="Options"
 JPREV="Prev"
 JPREVIOUS="Previous"
+JPREV_PAGE="Prev Page"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.ini:106]: Sort key in alpha order

**Comment by rdeutz** [comment]
Besides the question if the plugin should be removed, the markup has to be in a layout and not hardcoded within the plugin

**Comment by Quy** [comment]
Closing for reasons as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19303">issues.joomla.org/tracker/joomla-cms/19303</a>.</sub>

**Pull request closed without merging**